ocaml/testsuite/tests/parsetree/test.ml

```diff
@@ -79,31 +87,61 @@ let test parse_fun pprint print map filename =
             print_endline"====================================================="
           end
 
-let test parse_fun pprint print map filename =
-  try test parse_fun pprint print map filename
+let test parse_fun pprint print map filename ~extra_checks =
+  try test parse_fun pprint print map filename ~extra_checks
   with exn -> report_err exn
 
-let rec process path =
+let rec process path ~extra_checks =
   if Sys.is_directory path then
     let files = Sys.readdir path in
-    Array.iter (fun s -> process (Filename.concat path s)) files
+    Array.iter (fun s -> process (Filename.concat path s) ~extra_checks) files
   else if Filename.check_suffix path ".ml" then
     test
       Parse.implementation
       Pprintast.structure
       Printast.implementation
       (fun mapper -> mapper.Ast_mapper.structure)
       path
+      ~extra_checks
   else if Filename.check_suffix path ".mli" then
     test
       Parse.interface
       Pprintast.signature
       Printast.interface
       (fun mapper -> mapper.Ast_mapper.signature)
       path
+      ~extra_checks
+
+let process ?(extra_checks = fun _ -> Ok ()) text = process text ~extra_checks
+
+let check_all_extension_points_start_with text ~prefix =
+  let exception Unexpected_extension_point of string in
+  let check ~extension_point_prefix =
+    String.split_on_char '[' text
+    |> List.for_all (fun s ->
+        not (String.starts_with s ~prefix:extension_point_prefix)
+        ||   String.starts_with s ~prefix:(extension_point_prefix ^ prefix))
+  in
+  match
+    check ~extension_point_prefix:"%", check ~extension_point_prefix:"@"
```

Comment:
`check ~extension_point_prefix:"%" && check ~extension_point_prefix:"@"` is simpler, and does less work.  Whether you use an `if` or ` match ... with true -> ... | false -> ...` is up to you :-)

Reply:
Yeah, this was an artifact of an earlier refactor (see also the unused `let exception ... in` I deleted above :) ). Simplified.

Reply:
Congratulations for discovering the one time I'm okay with conditionals returning a boolean literal :-)